modules/programs/mpv.nix

```diff
@@ -49,30 +49,28 @@ let
   renderDefaultProfiles = profiles:
     renderOptions { profile = concatStringsSep "," profiles; };
 
-  mpvPackage = if cfg.scripts == [ ] then
-    pkgs.mpv
-  else
-    pkgs.wrapMpv pkgs.mpv-unwrapped { scripts = cfg.scripts; };
-
 in {
+  imports = [
+    (mkRemovedOptionModule [ "programs" "mpv" "scripts" ] ''
+      The `scripts` part was removed in favor of using `package` option
+      with `pkgs.wrapMpv`.
+
+      Replace `programs.mpv.scripts = [ ... ]` with
+      `programs.mpv.scripts = pkgs.wrapMpv pkgs.mpv-unwrapped { scripts = [ ... ]; }`.
+    '')
+  ];
+
   options = {
     programs.mpv = {
       enable = mkEnableOption "mpv";
 
       package = mkOption {
         type = types.package;
-        readOnly = true;
-        description = ''
-          Resulting mpv package.
-        '';
-      };
-
-      scripts = mkOption {
-        type = with types; listOf (either package str);
-        default = [ ];
-        example = literalExample "[ pkgs.mpvScripts.mpris ]";
+        default = pkgs.mpv;
+        example = literalExample
+          "pkgs.wrapMpv pkgs.mpv-unwrapped { scripts = [ pkgs.mpvScripts.mpris ]; }";
         description = ''
-          List of scripts to use with mpv.
+          Package providing mpv.
         '';
       };
```

Comment:
Biggest con of removing scripts is that module merging does not work for `types.package` so the user can no longer specify `programs.mpv.scripts` multiple times.

Reply:
But is this really a valid use case :thinking: ? I mean, I think it is bizarre for someone set the same option multiple times, and in the small amount of cases where this is desired there are other ways to do it (for example, a module where the user can override its own derivation).

Reply:
BTW, I would understand wanting to keep this option if the default was to have some scripts enabled by default, but this is not the case here.

Reply:
This is a genuine question, I am trying to think a case where the user will want to set `programs.mpv.scripts` with one content in one file and `programs.mpv.scripts` in another one (just to give an example).

Reply:
Ok, thought about one. If the user uses HM in both Linux and macOS, and wants to have some scripts available in both Linux and macOS and also `mpris` only in Linux it would be useful to have this behavior. I think it is pretty niche though, and of course will not work if you need to have, say, `vapoursynth` in Linux but not on macOS, but I think it is fair.